kf/kf_unix.go

```diff
@@ -247,45 +289,67 @@ func (b *BPF) LoadTCAttachProgram(ifaceName, direction string) error {
 				Kind: "clsact",
 			},
 		}
-
 		if err := tcgo.Qdisc().Add(&qdisc); err != nil {
 			log.Info().Msgf("could not assign clsact to %s : %v, its already exists", ifaceName, err)
 		}
-	}
 
-	bpfRootProg := b.ProgMapCollection.Programs[b.Program.EntryFunctionName]
+		if direction == models.IngressType {
+			parent = tc.HandleMinIngress
+		} else if direction == models.EgressType {
+			parent = tc.HandleMinEgress
+		}
+		progFD := uint32(bpfRootProg.FD())
+		// Netlink attribute used in the Linux kernel
+		bpfFlag := uint32(tc.BpfActDirect)
```

Comment:
Is it possible to relocate line 301 and 303 outside of the block in order to prevent duplicated code?

Reply:
Fixed in the newer commit